pkg/apis/v1/nodeclaim_conversion.go

```diff
@@ -94,7 +98,26 @@ func (in *NodeClaim) ConvertFrom(ctx context.Context, from apis.Convertible) err
 		return err
 	}
 	in.Annotations = lo.Assign(in.Annotations, map[string]string{KubeletCompatabilityAnnotationKey: kubeletAnnotation})
+	return in.setExpireAfter(ctx, v1beta1NC)
+}
 
+// only need to set expireAfter for v1beta1 to v1
+func (in *NodeClaim) setExpireAfter(ctx context.Context, v1beta1nc *v1beta1.NodeClaim) error {
+	kubeClient := injection.GetClient(ctx)
+	nodePoolName, ok := v1beta1nc.Labels[NodePoolLabelKey]
+	if !ok {
+		// If we don't have a nodepool for this nodeclaim, there's nothing to look up
+		return nil
+	}
+	nodePool := &NodePool{}
+	if err := kubeClient.Get(ctx, types.NamespacedName{Name: nodePoolName}, nodePool); err != nil {
```

Comment:
What do you think about adding an agnation that has the expireAfter value? That way we can make less call to the API server

Reply:
Reads are cached, no?

Reply:
That'a fair point.